### PR TITLE
Upgrade default python version to 3.8.4

### PR DESCRIPTION
**Pull request by ulgens**

### CODE REVIEW

**Comment by ulgens** [comment]
@thepwagner https://github.com/dependabot/dependabot-core/pull/1861#discussion_r427408949

On 3.8.3 upgrade PR, we resolved a linter issue with this solution but i'm not sure it's still feasible and have no idea how to split lines there. I added back `3.8.0` that line and i will be waiting for recommendations :slightly_smiling_face:

File: python/spec/dependabot/python/update_checker/pipenv_version_resolver_spec.rb [hunk 1]
```diff
@@ -255,7 +255,7 @@
                 to start_with("Dependabot detected the following Python")
               expect(error.message).to include("3.4.*")
               expect(error.message).
-                to include("supported in Dependabot: 3.8.3, 3.8.2, 3.8.1")
+                to include("supported in Dependabot: 3.8.4, 3.8.3, 3.8.2, 3.8.1, 3.8.0")
```
Comment (feelepxyz) [inline python/spec/dependabot/python/update_checker/pipenv_version_resolver_spec.rb:258]:
```suggestion
                to include("supported in Dependabot: 3.8.4, 3.8.3, 3.8.2, "\
                           "3.8.1, 3.8.0")
```

**Comment by feelepxyz** [comment]
@ulgens could you rebase on top of the latest `main` changes? Hopefully this will fix the tests

**Comment by ulgens** [comment]
@feelepxyz I did. Not sure why it failed this time.

**Pull request merged**